### PR TITLE
Rewrite PartitionStack data structure description

### PR DESCRIPTION
**Pull request by wilfwilson**

### CODE REVIEW

File: gap/partitionstack.gi [hunk 1]
```diff
@@ -2,27 +2,40 @@
...
-## cellsize: The length of cell i
-## fixed: A list of cells of size 1
-## splits: splits[i] is the cell which was split to create cell i
-##         (or -1 for splits[1])
+
+## invvals:   [4,5,2,3,1]  - The inverse of vals, to find where a value exists
+##                           in vals.
+## cellstart: [1,4,3]      - marks[cellstart[i]] = i.
+## cellsize:  [2,2,1]      - The length of cell i.
+## fixed:     [3]          - A list of cells of size 1. TODO in which order?
```
Comment (ChrisJefferson) [inline gap/partitionstack.gi]: In the order they became fixed. When a cell of size 2 is split into two cells of size 1, the original cell comes first.

File: gap/partitionstack.gi [hunk 2]
```diff
@@ -2,27 +2,40 @@
...
+## can be represented by the following pieces of information:
+
+## vals:   [5,3,4,1,2]    - A permutation of [1..n].
+## marks:  [1,0,3,2,0,6]  - marks[i] = j means cell 'j' starts at index 'i' of
+##                          vals. The n+1th element of the array is always n+1.
+## splits: [-1,1,2]       - splits[i] is the cell which was split to create
+##                          cell i (or -1 for splits[1])
 ## Note that:
-##   * The cells do not have to be in any particular order
+##   * The cells do not have to be in any particular order. TODO what?
```
Comment (ChrisJefferson) [inline gap/partitionstack.gi]: You can't assume the cells appear in the partition in order from smallest to largest, or anything like that. They appear (obviously) in practice just how they were split.

**Comment by wilfwilson** [comment]
Thanks for the comments @ChrisJefferson. I've adjusted the explanation and thereby removed the TODOs.

**Comment by wilfwilson** [comment]
Whoops, didn't mean to close

**Pull request merged**